README.md

```diff
@@ -21,33 +22,70 @@ This project provides a way to run an Apache Kafka cluster on Kubernetes and Ope
-For example it can use HostPath volumes on Minikube or Amazon EBS volumes in Amazon AWS deployments without any changes in the YAML files.
+This deployment uses the StatefulSets (previously known as "PetSets") feature of Kubernetes/OpenShift. 
+With StatefulSets, the pods receive a unique name and network identity and that makes it easier to identify the 
+individual Kafka broker pods and set their identity (broker ID). 
+Each Kafka broker pod is using its own PersistentVolume. 
+The PersistentVolume is acquired using PersistentVolumeClaim – that makes it independent on the actual type of the PersistentVolume. 
+For example, it can use HostPath volumes on Minikube or Amazon EBS volumes in Amazon AWS deployments without any changes in the YAML files.
 
-It's important to say that in this deployment both "regular" and "headless" services are used. The "regular" services are needed for having instances accessible from clients;
-the "headless" services are needed for having the DNS resolving directly the PODs IP addresses for having the Stateful Sets working properly.
+It's important to say that in this deployment both _regular_ and _headless_ services are used:
 
-This deployment is available under the _kafka-statefulsets_ folder and provides following artifacts :
+* regular services are needed for having instances accessible from clients outside the Kubernetes/OpenShift cluster;
+* headless services are needed to have DNS resolve the pods IP addresses directly, so that the StatefulSets work properly.
+
+This deployment is available under the _kafka-statefulsets_ folder and provides following artifacts:
 
 * Dockerfile : Docker file for building an image with Kafka and Zookeeper already installed
 * config : configuration file templates for running Zookeeper
 * scripts : scripts for starting up Kafka and Zookeeper servers
 * resources : provides all YAML configuration files for setting up volumes, services and deployments
 
-### Deploying to OpenShift
+### Deploying to OpenShift Origin
+
+1. With the [Openshift Origin client tools][origin] installed and having logged into your cluster, 
+   create the [provided "barnabas" template](kafka-statefulsets/resources/openshift-template.yaml) 
+   by running
+
+        oc create -f kafka-statefulsets/resources/openshift-template.yaml
+
+   in your terminal. This template provides the "zookeeper" StatefulSet with a single replica, the "kafka" StatefulSet with 3 replicas,
+   and the "zookeeper", "zookeeper-headless", "kafka" and "kafka-headless" Services.
+2. Create a new app using the "barnabas" template:
+
+        oc new-app barnabas
+
+### Deploying to Kubernetes
+
+1. If you don't have a cluster running, start one (e.g. by executing `minikube start`)
+2. Create the services by running:
+
+        kubectl apply -f kafka-statefulsets/resources/cluster-volumes.yaml && \
+        kubectl apply -f kafka-statefulsets/resources/zookeeper.yaml && \
```

Comment:
Point for discussion ... Do we want to keep the Kubernetes YAML files as separate files? Or do we want to bundle them all (apart from cluster-volumes.yaml) into single YAML file? That would make the deployment easier - it will be just a single command.

Reply:
I'm beginning to think a single file would make more sense. I'll open an issue to discuss it.

Reply:
Yeah, separate issue sounds like a good idea.